apps/payments/next/lib/utils/pageMetadata.ts

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+ import { headers } from 'next/headers';
+ import { fetchCMSData } from '@fxa/payments/ui/actions';
+ import { buildRedirectUrl } from '@fxa/payments/ui';
+ import { config } from 'apps/payments/next/config';
+ import type { Metadata } from 'next';
+ import type { CheckoutParams, } from '@fxa/payments/ui/server';
+
+  type Page =
+    | 'landing'
+    | 'new'
+    | 'start'
+    | 'success'
+    | 'error'
+    | 'location'
+    | 'page-not-found'
+    | 'processing'
+    | 'needs_input';
+  type PageType = 'checkout' | 'upgrade';
+
+ export async function makePageMetadata(
```

Comment:
Can you move this over to libs/payments/ui/src/lib/utils? Ideally, we keep as much business logic as we reasonably can out of the apps/payments/next directory.

As an added win, we can then import it like `import { makePageMetadata } from '@fxa/payments/ui'`, rather than the longer relative-path imports currently in use